src/tap_server/tap_server.py

```diff
@@ -0,0 +1,615 @@
+        <pre>
+curl -X POST http://localhost:5000/sync \\
+  -d "REQUEST=doQuery" \\
+  -d "LANG=ADQL" \\
+  -d "QUERY=SELECT TOP 10 ra, dec, mag FROM ztf_dr14 WHERE mag < 20"
+        </pre>
+
+        <h2>Supported ADQL Features</h2>
+        <ul>
+            <li>SELECT with column list or *</li>
+            <li>FROM with table name</li>
+            <li>WHERE clause with comparison operators</li>
+            <li>CONTAINS with POINT and CIRCLE for cone searches</li>
+            <li>TOP/LIMIT clause</li>
+        </ul>
+
+        <p><em>Note: This server uses the adql_to_lsdb module to
+               convert ADQL to LSDB operations.</em></p>
+    </body>
+    </html>
+    """
+    return html
+
+
+@app.route("/sync", methods=["GET", "POST"])
+def sync_query():
+    """
+    Synchronous query endpoint following TAP specification.
+
+    Accepts parameters:
+        REQUEST: Must be 'doQuery'
+        LANG: Query language (ADQL)
+        QUERY: The ADQL query string
+        FORMAT: Output format (default: votable)
+    """
+    # Get parameters from either GET or POST
+    params = request.form if request.method == "POST" else request.args
+
+    # Validate REQUEST parameter
+    request_type = params.get("REQUEST", "")
+    if request_type != "doQuery":
+        error_msg = f"Invalid REQUEST parameter: {request_type}. Must be 'doQuery'."
+        return Response(create_error_votable(error_msg), mimetype="application/xml", status=400)
+
+    # Validate LANG parameter
+    lang = params.get("LANG", "ADQL")
+    if lang.upper() != "ADQL":
+        error_msg = f"Unsupported query language: {lang}. Only ADQL is supported."
+        return Response(create_error_votable(error_msg), mimetype="application/xml", status=400)
+
+    # Get query
+    query = params.get("QUERY", "")
+    if not query:
+        error_msg = "Missing required parameter: QUERY"
+        return Response(create_error_votable(error_msg), mimetype="application/xml", status=400)
+
+    # Get format (default to votable)
+    output_format = params.get("FORMAT", "votable").lower()
+
+    try:
+        # Check if this is a TAP_SCHEMA query
+        if is_tap_schema_query(query):
+            # Query the TAP_SCHEMA metadata
+            data, result_columns = query_tap_schema(query)
+            # NOTE: Not sure this is quite right.  What comes back from caltech TAP?
+            table_name = "tap_schema"
+        else:
+            # Parse the ADQL query to get entities
+            entities = parse_adql_entities(query)
+
+            # Get the table name from the query
+            assert entities["tables"]
+            table = entities["tables"][0]
+
+            # Handle regular catalog query
+            # Convert table name like 'gaiadr3.gaia' to URL format
+            # catalog_prefix = "https://data.lsdb.io/hats"
+            # catalog_prefix = "http://epyc.astro.washington.edu:43210/hats"
+            # catalog_prefix = "/epyc/data3/hats/catalogs"
```

Comment:
Unsure if you want to keep this around for convenience in future iterations, or if these prefixes are no longer useful